### PR TITLE
Bump to wrangler v1.1.1 / Kubernetes 1.25 / ginkgo v2

### PR DESCRIPTION
**Pull request by brandond**
* Bump to wrangler v1.1.1 / Kubernetes 1.25 / ginkgo v2
* Tidy up Dockerfile.dapper to bump versions and do all the arch-specific stuff in one place.

There are ZERO code changes to the controller itself; this is all just updating to modern library versions and adapting tests to work with the newer upstream e2e framework changes.

### CODE REVIEW

**Review by VestigeJ — APPROVED** [review]
In my limited experience this looks good to me

**Comment by brandond** [comment]
@macedogm I just pulled in the dapper and sles->bci changes from your PR.

File: e2e/suite/plan_create_test.go [hunk 1]
```diff
@@ -1,6 +1,8 @@
 package suite_test
 
 import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
```
Comment (manuelbuil) [inline e2e/suite/plan_create_test.go:5]: In this file, you are adding these two imports but nothing else. I wonder if this is correct or a copy/paste?

File: e2e/suite/plan_resolve_test.go [hunk 2]
```diff
@@ -6,6 +6,8 @@ import (
 	"path"
 	"time"
 
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
```
Comment (manuelbuil) [inline e2e/suite/plan_resolve_test.go:10]: Same here

File: e2e/suite/channel_resolve_test.go [hunk 3]
```diff
@@ -5,6 +5,8 @@ import (
 	"net/http"
 	"net/http/httptest"
 
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
```
[the remark this answers is not in the document]
Comment (brandond) [reply e2e/suite/channel_resolve_test.go:9]: These are dot imports of the ginkgo and gomega DSLs. You'll find the same in the k3s and rke2 tests.

File: e2e/cluster/local/docker-compose.yaml [hunk 4]
```diff
@@ -6,17 +6,17 @@ volumes:
...
-  leader:
-    container_name: local-leader
+  server:
+    container_name: k3s-server-1
     image: "e2e/cluster/local/k3s"
     build:
       context: ./images/k3s/.
       args:
         - ARCH=${ARCH}
-    command: server --no-deploy=traefik,metrics-server
```
[the remark this answers is not in the document]
Comment (brandond) [reply e2e/cluster/local/docker-compose.yaml:16]: We were testing on a very old version of k3s!

File: e2e/suite/channel_resolve_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (brandond) [reply e2e/suite/channel_resolve_test.go:9]: You can also find examples at https://onsi.github.io/ginkgo/#bootstrapping-a-suite

[the remark this answers is not in the document]
Comment (manuelbuil) [reply e2e/suite/channel_resolve_test.go:9]: Ok, but there is still something weird: if there are ginkgo or gomega functions in this file, how did it work before if ginkgo and gomega were not being imported?

[the remark this answers is not in the document]
Comment (manuelbuil) [reply e2e/suite/channel_resolve_test.go:9]: By reading the link, I can see those files are generated by executing "ginkgo generate", so it must be correct then

**Review by manuelbuil — APPROVED** [review]

File: e2e/suite/channel_resolve_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (brandond) [reply e2e/suite/channel_resolve_test.go:9]: There were previously a bunch of DSL vars bound into another file in this package, you can see where I have removed them. That was the very old way of using it.

**Pull request merged**